vars/buildPluginWithGradle.groovy

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env groovy
+
+/**
+ * Simple wrapper step for building a plugin with Gradle.
+ */
+def call(Map params = [:]) {
+    // Faster build and reduces IO needs
+    properties([
+        durabilityHint('PERFORMANCE_OPTIMIZED'),
+        buildDiscarder(logRotator(numToKeepStr: '5')),
+    ])
+
+    def repo = params.containsKey('repo') ? params.repo : null
+    def failFast = params.containsKey('failFast') ? params.failFast : true
+    def timeoutValue = params.containsKey('timeout') ? params.timeout : 60
+    if(timeoutValue > 180) {
+      echo "Timeout value requested was $timeoutValue, lowering to 180 to avoid Jenkins project's resource abusive consumption"
+      timeoutValue = 180
+    }
+
+    boolean publishingIncrementals = false
+    boolean archivedArtifacts = false
+    Map tasks = [failFast: failFast]
+    buildPlugin.getConfigurations(params, true).each { config ->
```

Comment:
Should there be some filtering of configurations based on the fact that `javaLevel` and `jenkinsVersion` are not supported (at the moment)? Otherwise for some unsupported configurations a complete build would be performed with a misleading `stageIdentifier` and only a WARNING somewhere in the console log.

Reply:
It is better than the current behavior in `buildPlugin()` where the issues are silently ignored